src/macports1.0/sysctl.c

```diff
@@ -92,7 +91,37 @@ int SysctlCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int objc UNUSED,
     Tcl_SetObjResult(interp, tcl_result);
     return TCL_OK;
 #else
```

Comment:
We're assuming that all the other platforms we care about have `sysctlbyname`, then?

Reply:
Darwin (and maybe BSD?) should have `sysctlbyname`. The `else` case, at least in my testing, works on Linux which was my main focus. It [might](https://stackoverflow.com/a/150971) work on Solaris and AIX, although this hasn't been tested.

(Technically Linux does have sysctl, it's just [deprecated](https://www.phoronix.com/news/Linux-5.5-Kills-SYSCTL-SYSCALL) and behaves weirdly.)

Reply:
It's just that the else case is used on all platforms that don't have `sysctlbyname`. The Tcl code can handle sysctl failing, which is what used to happen on platforms we don't know how to implement sysctl on. Seems like now it would fail to compile in those cases.

Reply:
I'm not quite sure I understand. If sysctl fails, the else statement falls back to sysconf.  I tested this with an [ubuntu workflow](https://github.com/harens/macports-base/runs/7498881727), where it compiles successfully  (tests fail for an unrelated reason).

If sysconf and sysctl both fail, it should still compile and error successfully, just like before the PR. Rather than `Tcl_NewStringObj("sysctl not available", -1)`, it should produce `Tcl_NewStringObj("sysconf not available", -1)`. Please let me know if I've misunderstood :)

Reply:
I mean if there's a platform that doesn't have `sysctlbyname` and also doesn't have either `sysconf` or the `_SC_*` constants that are being used, the code will fail to compile, where previously it would compile and sysctl would just always fail.